### ISSUE TITLE
Investigate running different files in parallel

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/TSLint.MSBuild#51]
Imagine a numeric variable, `<TSLintParallelization>4</TSLintParallelization>`, that separates the list of TS files to compile into that many different sub-lists. Each list gets its own process.
* Upside: parallel linting!
* Downside: either nondeterministic result line orders or extra work done to sort them

**Comment by JoshuaKGoldberg** [comment]
Closing in favor of https://github.com/palantir/tslint/issues/2328.
